test/mocha.opts

```diff
@@ -1 +1,3 @@
+--compilers ts-node/register
```

Comment:
Did you try this without `ts-node/register`? Sucrase should generally work, and produces inline source maps (it actually always preserves line numbers, so source maps are trivial), but maybe it crashes nyc?

In any case, `ts-node/register` and `sucrase/register` both set up require hooks to compile typescript on the fly, so if we set up both hooks, one will just clobber the other, so we should pick one. I'd prefer for tests to be compiled with sucrase when run normally (I did some intentional work to switch to that), though if that's a pain, I'm fine switching back to ts-node.